bentoml/xgboost.py

```diff
@@ -225,21 +237,22 @@ def _setup_booster_params(
         return booster_params
 
     # pylint: disable=arguments-differ,attribute-defined-outside-init
-    def _setup(self) -> None:  # type: ignore[override]
+    def _setup(self) -> None:
         self._model = xgb.core.Booster(
             params=self._booster_params,
             model_file=self._model_file,
         )
         self._predict_fn = getattr(self._model, self._predict_fn_name)
 
-    # pylint: disable=arguments-differ
-    def _run_batch(  # type: ignore[override]
-        self, input_data: t.Union[np.ndarray, "pd.DataFrame", xgb.DMatrix]
-    ) -> "np.ndarray":
-        if not isinstance(input_data, xgb.DMatrix):
-            input_data = xgb.DMatrix(input_data)
-        res = self._predict_fn(input_data)
-        return np.asarray(res)
+    def _run_batch(
+        self, *args: t.Union[AnyNdarray, "DataFrame", xgb.DMatrix], **kwargs: str
+    ) -> AnyNdarray:
+        params = Params[t.Union[AnyNdarray, "DataFrame", xgb.DMatrix]](*args, **kwargs)
+        params = params.map(
+            lambda x: xgb.DMatrix(x) if not isinstance(x, xgb.DMatrix) else x
+        )
+        res = self._predict_fn(*params.args, **params.kwargs)
+        return t.cast(AnyNdarray, np.asarray(res))  # type: ignore[reportUnknownMemberType]
```

Comment:
why do we need t.cast here?

Reply:
asarray types is impartial, actually the cast is not necessary. I will just remove it